src/checkout-buttons/strategies/amazon-maxo/amazon-maxo-button-strategy.ts

```diff
@@ -0,0 +1,91 @@
+import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { AmazonMaxoPaymentProcessor, AmazonMaxoPlacement } from '../../../payment/strategies/amazon-maxo';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class AmazonMaxoButtonStrategy implements CheckoutButtonStrategy {
+    private _methodId?: string;
+    private _walletButton?: HTMLElement;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _amazonMaxoPaymentProcessor: AmazonMaxoPaymentProcessor
+    ) {}
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const { containerId, methodId } = options;
+
+        if (!containerId) {
+            throw new InvalidArgumentError('Unable to proceed because "containerId" argument is not provided.');
+        }
+
+        this._methodId = methodId;
+
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(() => this._amazonMaxoPaymentProcessor.initialize(this._getMethodId()))
+            .then(() => {
```

Comment:
This won't work because removing the braces will return the value of the `this._walletButton = this._createSignInButton(containerId))` expression, and the return value of the initialize method as is defined on [the interface implemented by this class](https://github.com/bigcommerce/checkout-sdk-js/blob/69c90fba41a307ea4f3077c853ba1f920c5b53c7/src/checkout-buttons/strategies/checkout-button-strategy.ts#L4)